lib/puppet/type/network_route_old.rb

```diff
@@ -0,0 +1,81 @@
+require 'ipaddr'
+require_relative '../../puppet_x/voxpupuli/utils.rb'
+
+Puppet::Type.newtype(:network_route) do
+  @doc = 'Manage non-volatile route configuration information'
+
+  include PuppetX::Voxpupuli::Utils
+
+  ensurable
+
+  newparam(:name) do
+    isnamevar
+    desc 'The name of the network route'
+  end
+
+  newproperty(:network) do
+    isrequired
+    desc 'The target network address'
+    validate do |value|
+      unless value == 'default'
+        a = PuppetX::Voxpupuli::Utils.try { IPAddr.new(value) }
+        raise("Invalid value for network: #{value}") unless a
+      end
+    end
+  end
+
+  newproperty(:netmask) do
+    isrequired
+    desc 'The subnet mask to apply to the route'
+
+    validate do |value|
+      unless value.length <= 3 || PuppetX::Voxpupuli::Utils.try { IPAddr.new(value) }
+        raise("Invalid value for argument netmask: #{value}")
+      end
+    end
+
+    munge do |value|
+      # '255.255.255.255'.to_i  will return 255, so we try to convert it back:
+      if value.to_i.to_s == value
+        # what are the chances someone is using /16 for their IPv6 network?
```

Comment:
I do that.

Reply:
That file is going away ;)

That said, I am looking for a way to transform a prefix length to subnet mask. Nothing seems to do it natively (plenty of libraries to transform from subnet mask to prefix length though).